src/IECoreImage/ImageReader.cpp

```diff
@@ -99,10 +100,24 @@ class ImageReader::Implementation
 
 			try
 			{
+
 				const ImageSpec *spec = m_cache->imagespec( m_inputFileName );
 
-				std::vector<float> data( spec->nchannels );
+				if( isDeep() )
+				{
+					DeepData deepData;
+					ImageInput *input = ImageInput::open( m_inputFileName.c_str() );
+					return input->read_native_deep_scanlines(
```

Comment:
Are there such thing as deep tiles? If so, I expect this would throw on a deep tiled image.